Update Go Version for Actions

crissi98 opened the pull request.

cniweb commented:
@crissi98 
https://github.com/HUK-COBURG/openapirouter/actions/runs/4079224885/jobs/7030368950#step:4:7

cniweb commented:
@crissi98 can you update the go version here:
https://github.com/HUK-COBURG/openapirouter/blob/1ff49e6f3bf60e58d5547187affa48a54ed98a66/.github/workflows/go.yml#L18
and here:
https://github.com/HUK-COBURG/openapirouter/blob/1ff49e6f3bf60e58d5547187affa48a54ed98a66/.github/workflows/shiftleft.yml#L22

crissi98 commented:
@cniweb I updated the go version for ShiftLeft as well. Additionally I added a pipeline step to download the dependencies before building the app. This missing step was the actual cause for the failure of the ShiftLeft action
